internal/middleware/hmac_sign.go

```diff
@@ -0,0 +1,124 @@
+package middleware
+
+import (
+    "bytes"
+    "crypto/hmac"
+    "crypto/sha256"
+    "encoding/hex"
+    "errors"
+    "go.uber.org/zap"
+    "hash"
+    "io"
+    "net/http"
+)
+
+const SignedBodyHeader = "HashSHA256"
+
+func SignBody(key string, log *zap.Logger) func(next http.Handler) http.Handler {
+    if key == "" {
+        log.Debug("body sign middleware disabled. no key found")
+    }
+
+    return func(next http.Handler) http.Handler {
+        if key == "" {
+            return next
+        }
+
+        fn := func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Нет смысла создавать отдельную локальную перменную, проще сделать так:


```suggestion
        return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```